Userland/Shell/main.cpp

```diff
@@ -104,38 +101,26 @@ int main(int argc, char** argv)
     parser.add_positional_argument(file_to_read_from, "File to read commands from", "file", Core::ArgsParser::Required::No);
     parser.add_positional_argument(script_args, "Extra arguments to pass to the script (via $* and co)", "argument", Core::ArgsParser::Required::No);
 
-    parser.parse(argc, argv);
+    parser.parse(arguments);
 
     if (format) {
-        auto file = Core::File::open(format, Core::OpenMode::ReadOnly);
-        if (file.is_error()) {
-            warnln("Error: {}", file.error());
-            return 1;
-        }
+        auto file = TRY(Core::File::open(format, Core::OpenMode::ReadOnly));
 
         initialize();
 
         ssize_t cursor = -1;
-        puts(shell->format(file.value()->read_all(), cursor).characters());
+        puts(shell->format(file->read_all(), cursor).characters());
         return 0;
     }
 
     auto pid = getpid();
     if (auto sid = getsid(pid); sid == 0) {
-        if (setsid() < 0) {
-            perror("setsid");
-            // Let's just hope that it's ok.
-        }
+        TRY(Core::System::setsid());
```

Comment:
This does not have the same semantics.

Reply:
As far as I understand, the only difference is that it was allowed to fail and now this is an error.
Is it an issue by itself, or the problem is that it's not justified and in another PR?

Anyway, after reading some POSIX rules, I still don't understand why it is allowed to fail, and so why shouldn't we enforce it. Can you shed some light on?

Reply:
This can fail for a few rather surprising reasons:
- someone setsid()'d in our process before exec()'ing us (iirc bash did this in some version)
- we just happened to have a PID that's equal to an existing PGID because of PID rollover in a long-running system

This aside though, a "port to libmain" commit should not change program semantics anyway, that's not very atomic now is it? :P

Reply:
I should have mentioned that I realized that indeed it hasn't a place in this PR and that I waited to know if I should create a new PR.
These are obviously surprising reasons of why it can fail, but not why we don't enforce the creation of a new session.
In both case, if we `fork()` before (it might be an awful idea, I'm really not sure), we should be confident enough to ensure the success of the syscall.

Thank you for your answer.
I'm really happy to learn new things thanks to this project!

Reply:
It being in this PR isn't a problem at all, it should just be in a separate commit :)

Forking first thing in the startup process? Is having made our own session that important?
If you think about it, setsid only fails as either an accident, or if we already have what we want, I'm not sure why we need to go out of our way to accommodate for those, why not just let it fail?

Reply:
I thought that it was necessary to ensure to have the correct tty, but I will totally agree with you as I don't know any details about it.
As my final argument: because we could use `TRY()` and that's cool.